tools/aqrdev

```diff
@@ -200,26 +200,52 @@ def cmd_create(
         click.secho(f"Deployment '{name}' already exists", fg="red")
         sys.exit(errno.EEXIST)
 
+    """
+    User did not assign box or image
+    1) search for existing default aquarium box and provider type, then flag it
+    2) look up available image name and provider type, then flag it 
+    3) if neither exist, error out
+    """
     if not box and not image:
-        if "aquarium" in avail_boxes:
-            box = "aquarium"
-        elif avail_images and "aquarium" in avail_images:
-            image = "aquarium"
-        elif name in avail_boxes:
-            box = name
-        elif avail_images and name in avail_images:
-            image = name
+        click.secho(f"Using default 'aquarium' as box name and image name", fg="green")
+        aquarium_box_exist = False
+        aquarium_image_exist = False
+        for box_name, provider_name in avail_boxes:
+            if box_name == "aquarium" or box_name == name:
+                aquarium_box_exist = True
+                box = box_name
+                provider = provider_name
+        for img_obj in avail_images:
+            if aquarium_box_exist:
+                if img_obj.name == box and img_obj.provider == provider:
+                    aquarium_image_exist = True
+            elif img_obj.name == "aquarium" or img_obj.name == name and img_obj.provider == provider:
+                aquarium_image_exist = True
+                image = img_obj.name
+                provider = img_obj.provider
+
+        if aquarium_box_exist:
+            click.secho(f"Box '{box}' exist with Provider '{provider}' is availble", fg="green")
+        elif aquarium_image_exist:
+            click.secho(f"Image '{image}' exist with Provider '{provider}' is availble", fg="green")
         else:
-            click.secho("Unable to find default box 'aquarium'", fg="red")
+            click.secho("Unable to find default box or default image 'aquarium'", fg="red")
             sys.exit(errno.ENOENT)
 
+    """
+    Either User assign the image name or we get it from listing existing image
+    1) check if user assign image still exist
+    2) if image name is box name with force flag, remove the box
+    3) create box with image name, if image already in box list with no force flag, reuse
+    """
     if not box and image:
```

Comment:
It would be enough to check for image only, if we move both box and image presence check to the beginning of this block. Btw, the comment to above is also very difficult to read.

Reply:
@kshtsk the original design is if image exist then use it. If force flag remove the old one use the new one. If no force then use the existing one. You need to understand the original design of the code to review the logic. I didn't change any logic at all.